nautobot/extras/factory.py

```diff
@@ -140,6 +142,23 @@ def created(self):
         return faker.Faker().past_datetime(start_date=self.job_result.date_created, tzinfo=timezone.utc)
 
 
+class JobQueueFactory(PrimaryModelFactory):
+    """JobQueue model Factory"""
+
+    class Meta:
+        model = JobQueue
+
+    class Params:
+        is_celery_type = NautobotBoolIterator()
+        has_description = NautobotBoolIterator()
+        has_tenant = NautobotBoolIterator()
+
+    name = factory.LazyAttributeSequence(lambda o, n: f"{o.queue_type} Job Queue - {n}")
+    description = factory.Maybe("has_description", factory.Faker("text", max_nb_chars=CHARFIELD_MAX_LENGTH), "")
+    queue_type = factory.Maybe("is_celery_type", JobQueueTypeChoices.TYPE_CELERY, JobQueueTypeChoices.TYPE_KUBERNETES)
```

Comment:
Perhaps?

```suggestion
    queue_type = factory.Iterator(JobQueueTypeChoices.CHOICES, getter=lambda choice: choice[0])
```

Would eliminate the need for a `is_celery_type` flag at all and will automatically handle future cases when we add to JobQueueTypeChoices?